### PR TITLE
Correcting a call to Put

### PR DESCRIPTION
**Pull request by richev-o**
Fixed a call to `Put` to include a resource.

### CODE REVIEW

File: source/Octopus.Server.Client/Repositories/Async/DeploymentSettingsRepository.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ public async Task<DeploymentSettingsResource> Get(ProjectResource project)
                 throw new NotSupportedException(
                     $"Version Controlled projects are still in Beta. Use {nameof(IDeploymentSettingsBetaRepository)}.");
 
-            await client.Put(deploymentSettings.Link("Self"));
+            await client.Put(deploymentSettings.Link("Self"), project);
```
Comment (MarkSiedle) [inline source/Octopus.Server.Client/Repositories/Async/DeploymentSettingsRepository.cs:47]: @YuKitsune has a different fix for this on this branch atm: https://github.com/OctopusDeploy/OctopusClients/pull/588/files/72ea3f111ffc59839b0049cabcc83d9530c300d0..e95ebe83e87a0ce87498b90ac7420e5fd2f1d5b3

I think this should have been sending through the deploymentSettings object as part of the PUT?

Comment (YuKitsune) [reply source/Octopus.Server.Client/Repositories/Async/DeploymentSettingsRepository.cs:47]: That right, it should be the deploymentSettings, my branch has some other fixes related to wrong links being used too.

Comment (richev-o) [reply source/Octopus.Server.Client/Repositories/Async/DeploymentSettingsRepository.cs]: Have reworked the change, with help from @dylanlerch

**Review by MarkSiedle — APPROVED** [review]
Ah yes, the `DeploymentSettings` link from earlier today :) Thx all. Looks good.

**Pull request merged**